scripts/sct_compute_hausdorff_distance.py

```diff
@@ -384,10 +385,19 @@ def resample_image(fname, suffix='_resampled.nii.gz', binary=False, npx=0.3, npy
         if binary:
             interpolation = 'nn'
 
+        if nz == 1:  # when data is 2d: we convert it to a 3d image in order to avoid nipy problem of conversion nifti-->nipy with 2d data
+            sct.run('sct_image -i ' + ','.join([fname, fname]) + ' -concat z -o ' + fname)
+
         sct.run('sct_resample -i ' + fname + ' -mm ' + str(npx) + 'x' + str(npy) + 'x' + str(pz) + ' -o ' + name_resample + ' -x ' + interpolation)
 
+        if nz == 1:  # when input data was 2d: re-convert data 3d-->2d
+            sct.run('sct_image -i ' + name_resample + ' -split z')
+            im_split = Image(name_resample.split('.nii.gz')[0] + '_Z0000.nii.gz')
+            im_split.setFileName(name_resample)
+            im_split.save()
+            del im_split
```

Comment:
I think you can let the garbage collection take care of the deletion here, did you faced issues with memory ? If not you can remove this line.

Reply:
Right! Done!